### PR TITLE
Run ROPgadget with subprocess.Popen

### PR DESCRIPTION
**Pull request by nurmukhametov**
I encountered a problem during support exrop in rop-benchmark (https://github.com/ispras/rop-benchmark/issues/1). Benchmark environment contains broken /bin/sh (which points to /zh script, zh script does nothing except writing some messages to console), it is needed to check the exploit workability.

The function `popen` calls ROPgadget as `/bin/sh -c ROPgadget`. There are several possible solutions to work around this. I think the easiest way to use `subprocess.Popen` and call ROPgadget directly avoiding implicit /bin/sh invocation.

### CODE REVIEW

File: Exrop.py [hunk 1]
```diff
@@ -5,18 +5,28 @@
...
-        datas.sort(key=len) # sort by length
-        for data in datas:
-            addr,insns = data.split(" : ")
-            insstr,opcode_hex = insns.split(" // ")
-            opcode = bytes.fromhex(opcode_hex)
-            addr = int(addr, 16)
+    from subprocess import Popen, PIPE, STDOUT
+    import re
+
+    cmd = ['ROPgadget', '--binary', filename, '--multibr', '--only',
```
Comment (d4em0n) [inline Exrop.py:11]: this changes breaks the feature where the user can add additional ropgadget options (i will document it later), please add `opt` variable before `--binary`

**Comment by nurmukhametov** [comment]
It's been added on two lines later.

File: Exrop.py (hunk 1 above)
Comment (nurmukhametov) [reply Exrop.py:11]: It's been added on two lines later.

**Comment by d4em0n** [comment]
thanks!

**Pull request merged**